### ISSUE TITLE
agent.setConfiguration() and agent.setState() / agent.setStatus() are not working in latest versions(1.7.5,1.7.0) which is working fine in lower versions(1.6.1, 1.6.3)

### ISSUE THREAD

**Issue by tamilmani-m** [amazon-connect/amazon-connect-streams#516]

**Comment by tamilmani-m** [comment]
Below are the issues which was identified 
**Version 1.7.5, 1.7.0**
agent.setConfiguration()
var config = agent.getConfiguration();
config.extension = "+18005550100";
config.softphoneEnabled = false;
agent.setConfiguration(config {
   success: function() { /* ... */ },
   failure: function(err) { /* ... */ }
});
**Issue:** InvalidLocale

agent.setState() / agent.setStatus()
var state = agent.getAgentStates()[0];
agent.setState(state, {
    success: function() { /* ... */ },
    failure: function(err) { /* ... */ },
   },
   {enqueueNextState: false}
});
**Issue:** {"type":"UnexpectedParameter","message":"Unexpected key 'enqueueNextState' found in params","stack":["UnexpectedParameter: Unexpected key 'enqueueNextState' found in params","    at ParamValidator.fail (https://connect-config-dev.my.connect.aws/static/connect-stream.js:6514:37)","    at ParamValidator.validateStructure (https://connect-config-dev.my.connect.aws/static/connect-stream.js:6541:14)","    at ParamValidator.validateMember (https://connect-config-dev.my.connect.aws/static/connect-stream.js:6552:21)","    at ParamValidator.validate (https://connect-config-dev.my.connect.aws/static/connect-stream.js:6498:10)","    at Request.VALIDATE_PARAMETERS (https://connect-config-dev.my.connect.aws/static/connect-stream.js:4957:42)","    at Request.callListeners (https://connect-config-dev.my.connect.aws/static/connect-stream.js:8768:20)","    at callNextListener (https://connect-config-dev.my.connect.aws/static/connect-stream.js:8758:12)","    at Request.VALIDATE_CREDENTIALS (https://connect-config-dev.my.connect.aws/static/connect-stream.js:4911:53)","    at Request.callListeners (https://connect-config-dev.my.connect.aws/static/connect-stream.js:8764:18)","    at Request.emit (https://connect-config-dev.my.connect.aws/static/connect-stream.js:8740:10)"]}

**Version 1.6.8**
agent.setConfiguration()
var config = agent.getConfiguration();
config.extension = "+18005550100";
config.softphoneEnabled = false;
agent.setConfiguration(config {
   success: function() { /* ... */ },
   failure: function(err) { /* ... */ }
});
**Issue:** ERROR ReferenceError: AgentErrorStates is not defined
    at Agent.push.3168.Agent.setConfiguration (connect-streams.js:24158)
    
  **Version 1.6.3**
    loginPopup: true,               // optional, defaults to `true`
    loginPopupAutoClose: true,      // optional, defaults to `false`
    loginOptions: {                 // optional, if provided opens login in new window
      autoClose: true,              // optional, defaults to `false`
      height: 600,                  // optional, defaults to 578
      width: 400,                   // optional, defaults to 433
      top: 0,                       // optional, defaults to 0
      left: 0                       // optional, defaults to 0
    },
  **Issue:** Login popup is not opening for specified height and weight Rather it's opening in new tab 
  
  **Please help on this**

**Comment by vpogudina** [comment]
Hi tamilmani-m, I'm checking the snippets you provided:
for the issue: InvalidLocale you will need to set a locale in config as well like this config.agentPreferences.locale='en_US' (or another valid locale string)

Regarding this one:
```
agent.setState() / agent.setStatus()
var state = agent.getAgentStates()[0];
agent.setState(state, {
success: function() { /* ... / },
failure: function(err) { / ... */ },
},
{enqueueNextState: false}
});
```
You have extra } at the end, it should be just );

**Comment by seiyako** [comment]
Hi @tamilmani-m, for the setConfiguration issue, this is apparently a bug. Please see the workaround posted to another issue #460

**Comment by seiyako** [comment]
fixed in v2.2.0

see https://github.com/amazon-connect/amazon-connect-streams/issues/460
